hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -1305,11 +1525,48 @@ Connection::close(const std::string& reason, std::exception_ptr cause)
     });
 }
 
+void
+Connection::set_tpc_channels(std::vector<std::unique_ptr<socket>> tpc_channels)
```

Comment:
It is called with an rvalue "connection_->set_tpc_channels(move(tpc_channels_));". As it is just used in this place, it might be better to make the parameter as an rvalue to decrease copy.

Reply:
I don't want to make it more complex. There will be probably `16` or `32` cores approximately. So trying to prevent cost of one extra `move` operation will probably not affect the performance, probably it will be eliminated in compile time. Also taking parameter by value is far simpler than taking any type of reference. So I generally prefer to stay in safe side unless there is a practical reason not to.